src/main/kotlin/de/itemis/mps/gradle/generate/Plugin.kt

```diff
@@ -0,0 +1,92 @@
+package de.itemis.mps.gradle.generate
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import java.io.File
+
+
+data class Plugin(
+        var id: String,
+        var path: String
+)
+
+data class Macro(
+        var name: String,
+        var value: String
+)
+
+open class GeneratePluginExtensions {
+    lateinit var mpsConfig: Configuration
+    var mpsLocation: File? = null
+    var plugins: List<de.itemis.mps.gradle.generate.Plugin> = emptyList()
+    var pluginLocation: File? = null
+    var models: List<String> = emptyList()
+    var macros: List<Macro> = emptyList()
+    var projectLocation: File? = File("./mps-prj")
+    var debug = false
+}
+
+open class GenerateMpsProjectPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
+        project.run {
+
+            val extension = extensions.create("generate", GeneratePluginExtensions::class.java)
+            val mpsLocation = extension.mpsLocation ?: File(project.buildDir, "mps")
+
+            val genConfig = configurations.maybeCreate("genRuntime")
```

Comment:
Have you considered using a detached (anonymous) configuration instead of a named one? https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/ConfigurationContainer.html#detachedConfiguration-org.gradle.api.artifacts.Dependency...-

Reply:
I wasn't sure here which one to use. The named one might have the advantage that the user could use it in a dependency resolution strategy block and force a specific version. While the detached one is nice because it would conflicts with anything the user has created.

Reply:
I would use a detached config and let the user specify the dependency, similar to how JDK is selected in the CreateDmg task.

Reply:
Done.